spec/spec_helper_acceptance.rb

```diff
@@ -29,35 +29,65 @@
+          elsif host['platform'] =~ /^ubuntu|^debian/
+            on(host, 'apt-get install -y zlib1g-dev patch g++')
+          end
 
-      path = host.file_exist?("#{host['privatebindir']}/gem") ? host['privatebindir'] : host['puppetbindir']
-      on(host, "#{path}/gem install azure_mgmt_compute azure_mgmt_network azure_mgmt_resources azure_mgmt_storage hocon retries azure")
-    end
-  end
+          gem_command = if host['platform'] =~ /^windows/
+                          'cmd.exe /c cmd.exe /c "C:\Program Files\Puppet Labs\puppet\sys\ruby\bin\gem.bat"'
+                        else
+                          host.file_exist?("#{host['privatebindir']}/gem") ? "#{host['privatebindir']}/gem" : "#{host['puppetbindir']}/gem"
+                        end
+          if host['platform'] =~ /^windows/
+            # Unf has a separate gem for it's native extensions, unf_ext. Unf_ext has windows packages with the precompiled
+            # libraries. Because of this setup just installing the top level azure gems doesn't always seem to do the
+            # right thing on Windows
+            on(host, "#{gem_command} install unf")
+          end
+          on(host, "#{gem_command} install azure_mgmt_compute azure_mgmt_network azure_mgmt_resources azure_mgmt_storage hocon retries azure")
+        end
+      end
 
-  proj_root = File.expand_path(File.join(File.dirname(__FILE__), '..'))
-  hosts.each do |host|
-    # set :target_module_path manually to work around beaker-rspec bug that does not
-    # persist distmoduledir across runs with reused nodes
-    # TODO: ticket up this bug for beaker-rspec
-    install_dev_puppet_module_on(host, :source => proj_root, :module_name => 'azure', :target_module_path => '/etc/puppetlabs/code/modules')
-  end
+      proj_root = File.expand_path(File.join(File.dirname(__FILE__), '..'))
+      hosts.each do |host|
+        # set :target_module_path manually to work around beaker-rspec bug that does not
+        # persist distmoduledir across runs with reused nodes
+        # TODO: ticket up this bug for beaker-rspec
+        if host['platform'] =~ /^windows/
+          install_dev_puppet_module_on(host, :source => proj_root, :module_name => 'azure', :target_module_path => '/cygdrive/c/ProgramData/PuppetLabs/code/modules')
+        else
+          install_dev_puppet_module_on(host, :source => proj_root, :module_name => 'azure', :target_module_path => '/etc/puppetlabs/code/modules')
+        end
+      end
 
-  # Deploy Azure credentials to all hosts
-  if ENV['AZURE_MANAGEMENT_CERTIFICATE']
-    hosts.each do |host|
-      scp_to(host, ENV['AZURE_MANAGEMENT_CERTIFICATE'], '/tmp/azure_cert.pem')
+      # Deploy Azure credentials to all hosts
+      if ENV['AZURE_MANAGEMENT_CERTIFICATE']
+        if default['platform'] =~ /^windows/
+          scp_to(default, ENV['AZURE_MANAGEMENT_CERTIFICATE'], WINDOWS_AZURE_CERT)
+        elsif default['platform'] =~ /^ubuntu|^debian|^el/
```

Comment:
this leaves the option open for this to run through for platforms that don't match, but for which we don't upload the management cert. Given that will then leads to failures I would either:

* Throw an error in an else clause
* Remove the elsif part and just use else, assuming all other platforms will be linux-like